veg2hab/vegkartering.py

```diff
@@ -671,31 +673,48 @@ class Kartering:
     ]
 
     def __init__(self, gdf: gpd.GeoDataFrame):
-        # TODO clean this up!
-        try:
-            self.gdf = gdf[
-                self.PREFIX_COLS + self.HABTYPE_COLS + self.POSTFIX_COLS
-            ].copy()
-        except KeyError:
+        assert (
+            len(gdf._state.unique()) == 1
+        ), "Alle vlakken moeten dezelfde state hebben"
+        state = gdf._state.iloc[0]
+
+        if state in [KarteringState.PRE_WWL, KarteringState.POST_WWL]:
             self.gdf = gdf[
                 self.PREFIX_COLS + self.VEGTYPE_COLS + self.POSTFIX_COLS
             ].copy()
 
-        if not self.gdf["ElmID"].is_unique:
-            raise ValueError("ElmID is niet uniek")
-
-        # Als HabitatKeuze wel bestaat dan hoeven we VegTypeInfos niet te sorteren;
-        # Deze zijn dan al op de juiste volgorde (namelijk die van de HabitatKeuzes)
-        if not "HabitatKeuze" in self.gdf.columns:
             # Alle VegTypeInfo sorteren op percentage van hoog naar laag
-            # (Dit voornamelijk omdat dan als bij de mozaiekregels v0.1 we overal de eerste habitatkeuze
-            #  als enige habitatkeuze nemen, we altijd de habitatkeuze met het hoogste percentage nemen)
             self.gdf["VegTypeInfo"] = self.gdf["VegTypeInfo"].apply(
                 lambda x: sorted(x, key=lambda y: y.percentage, reverse=True)
             )
 
-        # NOTE: evt iets van self.stage = lokaal/sbb/vvn ofzo? Enum?
-        #       Misschien een dict met welke stappen gedaan zijn?
+        elif state in [KarteringState.MITS_HABKEUZES, KarteringState.MOZAIEK_HABKEUZES]:
+            self.gdf = gdf[
+                self.PREFIX_COLS + self.HABTYPE_COLS + self.POSTFIX_COLS
+            ].copy()
+
+        else:
+            raise ValueError(
+                f"Instantieren kartering met state {state} is niet ondersteund"
+            )
+
+        if not self.gdf["ElmID"].is_unique:
+            raise ValueError("ElmID is niet uniek")
+
+    def has_state(self, *states: KarteringState) -> bool:
```

Comment:
Ik zou hier denk ik check_state van maken en deze ook meteen de value error laten raisen.

Reply:
Klopt deze typehint zo, of moet dat een list zijn? Ik heb eigenlijk geen idee.

Reply:
Ik vind tegenstrijdige dingen op internet (https://stackoverflow.com/a/37032111 zegt `*args: int` maar https://www.slingacademy.com/article/python-typing-a-function-with-args-and-kwargs/ zegt `*args: Tuple[int, ...]` en in de docs staat hier https://docs.python.org/3/library/typing.html `*args: int` als mogelijkheid...)

ChatGPT zegt `*args: int` maar als je explicieter wilt zijn kan  `*args: Tuple[int, ...]` ook. Het is en blijft dus een mysterie.

Reply:
> Ik zou hier denk ik check_state van maken en deze ook meteen de value error laten raisen.

done :)